doc/lsst.ip.diffim/code_notes.rst

```diff
@@ -0,0 +1,183 @@
+#################################
+Package usage and technical notes
+#################################
+
+This page is a collection of usage and code related notes about the
+image differencing implmentation. We do not summarise the Alard-Lupton
+(AL) [AL_1998]_ and Zackay, Ofek, Gal-Yam (ZOGY) [ZOGY2016]_ papers
+themselves here. 
+
+
+Usage notes
+===========
+
+Supported input data
+--------------------
+
+ImageDifferenceTask supports two types of data products as templates,
+*coadds* and *calexps*, which can be subtracted from calexp science
+exposures *only*. Other combinations of inputs are not supported.  The
+input mode is selected by the ``getTemplate`` configurable field.  The
+``getTemplate`` subtask is a properly retargetable top level
+``pexConfig.ConfigurableField`` with two supported target subtasks:
+``GetCoaddAsTemplate`` and ``GetCalexpAsTemplate``.
+
+Enabling processing steps
+-------------------------
+
+There is a sequence of pre-subtraction and post-subtraction processing
+steps included in ``ImageDifferenceTask`` around the actual
+subtraction operation of the images. Including the subtraction
+operation itself, these steps can be enabled or disabled by top level
+``do<ACTION>`` configuration options. These top-level configuration
+options are summarised in the `subtasks flowchart
+<https://github.com/lsst-dm/diffimTests/tree/master/figure_subtasks>`_
+. Some of these top level configuration options are also passed on to
+invoked subtasks and influence their functionality. They may not be
+specified for the subtasks directly.
+
+Specifying a coadd as template
+------------------------------
+
+Using the ``GetCoaddAsTemplate`` subtask, we specify one or more
+science exposures by the ``--id`` general data specification
+option. Tract or patch specification as part of the ``--id`` option
+are silently ignored. The given (``coaddName``, defaults to "deep")
+type coadd is looked up in the same repository automatically. The
+subtask detemines the sky corners of a box that fully contains the
+science exposure with a configurable boundary
+(``templateBorderSize``). The subtask selects all patches of the coadd
+skymap that overlap with the science exposure and cuts them
+accordingly to assemble the template coadd image. The template coadd
+image inherits the coadd WCS. It is an error if there is no
+overlapping region. The ``--templateId`` option is silently
+ignored. The following example specifies one science exposure and
+implicitly the deep coadd residing in the same repository for image
+differencing:
+
+.. code-block:: python
```

Comment:
I think these code blocks should be bash rather than python, since they are things you type on the command line rather than into a python interpreter.

Reply:
Done.